ament_index_cpp/include/ament_index_cpp/get_resource.hpp

```diff
@@ -64,6 +68,20 @@ get_resource(
   std::string & content,
   std::string * prefix_path = nullptr);
 
+/// Get a the content and path of a resource
+/**
+ * \param[in] resource_type type of the resource
+ * \param[in] resource_name name of the resource
+ * \return a pair with the the installation prefix of the given resource if found
+ *         and the contents of the resource
+ * \throws std::runtime_error if resource_type or resource_name are empty.
+ */
+AMENT_INDEX_CPP_PUBLIC
+std::pair<std::optional<std::filesystem::path>, std::string>
```

Comment:
I am not the biggest fan of std::pair as return type.
A small struct like

```suggestion
struct PathWithResource
{
std::optional<std::filesystem::path> resourcePath;
// ?
std::string contents;
}
```
is nicer and makes the code using the return type way more understandable.

Also from the documentation I can't figure out what contents is / should contain.